container_pipeline/cleanup_registry/garbagecollector.py

```diff
@@ -136,33 +91,31 @@ def _mark_for_removal(self):
         """Orphans mismatched images from registry."""
         lib.print_msg("Marking mismatched containers for removal...",
                       self._verbose)
-        registry_storage_path = "/var/lib/registry/docker/registry/v2"
-        registry_blobs = registry_storage_path + "/blobs"
-        registry_repositories = registry_storage_path + "/repositories"
         for container_full_name, tag_list in self._mismatched.iteritems():
             # For every entry in mismatched, if a build is not currently running
             # remove it
             ## Formulate necessary data
-            namespace = container_full_name.split("/")[0] \
-                if "/" in container_full_name else container_full_name
-            namespace_path = registry_repositories + "/" + namespace
-            container_name = registry_repositories + "/" + container_full_name
-            manifests = container_name + "/_manifests"
-            tags = manifests + "/tags"
-            # Delete the tag
-            for item in tag_list:
-                container_tag = str.format(
-                    "{container_name}/{container_tag}",
-                    container_name=container_full_name,
-                    container_tag=item
+            for tag in tag_list:
+                if "/" in container_full_name:
+                    container_namespace, container_name = container_full_name.split(
```

Comment:
we are excluding the library images here?

Reply:
No, we arent. Library images do get cleaned up too.  Tested on my setup.